### DISCUSSION TITLE
Table Row Button to open form with table row details

### DISCUSSION THREAD

**Discussion by Dretreyt** [Budibase/budibase#14558] [Help] [answered]
Hi,

Apologies if this is a rather simple question, but I am stuck.

I have a create record form working okay, which will add records to a table all okay, and i have a table that can list all records in the database table, this budibase table has a button on the end that I want to be able to click on and populate a form which will let me edit the record values. I have parsed the value i want (prefixed_referralID) to the title of a Card and i can see it change depending on the button i click on. However the form doesnt populate with the data.

My current screens are:

/xxx/edit:id      -> this is the page that i need to load with the form on to edit the row selected data
/xxx/list        ->  this is the table with all records from the database
/xxx/new    ->  this is the create record table

the button on the '/xxx/list' page has a Navigate To action to 'Screen' /xxx/edit/ {{ New Table.xxx.prefixed_referralID }}

/xxx/edit/:id    Data Provider, Repeater and a card, the Card Title does change to the prefixed_referralID depending on the recordi Click but the form doesnt populate with data.

Any help would be greatly appreciated

**Comment by mikesealey** [comment, marked as answer]
Hey @Dretreyt 

_"i have a table that can list all records in the database table, this budibase table has a button on the end that I want to be able to click on and populate a form which will let me edit the record values."_

In the `On Click` actions, use an "Update State" action to set the name of the state to be `prefixed_referralID`, and the value to be the `_id`. (I'm using sample data)
![image](https://github.com/user-attachments/assets/d4b77f36-1b2c-4a5e-b5a3-0897488c3c9f)

Then, wherever you want to update that record, use a Form Block, set to Update, and pass it the state you've just declared `{{ State.prefixed_referralID }}`
![image](https://github.com/user-attachments/assets/73138269-1a23-409a-9351-ec5423328c98)

Each time a different row is clicked, that row's ID is set in state, which will update the information in the Form Block.

**Comment by Dretreyt** [comment]
Hi Mike,

Absolute legend, thanks for your help on this.

Kind regards

Rob

Robert Burt
NEC Software Solutions
Bexley Civic Offices
Tel: 020 3045 3793
Ext: 3793

Annual Leave:
